### ISSUE TITLE
[xenial] Keyboard opacity when set around 80% is not consistent

### ISSUE THREAD

**Issue by ivoxavier** [ubports/ubuntu-touch#586]
- Device: Fairphone 2
- Channel: 16.04/devel
- Build: 273

### Steps to reproduce
1-Set opacity keyboard around 80%
2- Go to Security and Privacy
3-Go to Blocking and unlocking
4-Go to Lock security and set 4-digit passcode or passphrase

### Expected behavior
Consistent UI for keyboard

### Actual behavior
Half broken ?? :)

![photo_2018-05-08_14-05-31](https://user-images.githubusercontent.com/16002474/39759435-95f6cba0-52ca-11e8-995f-71c323940876.jpg)

**Comment by NeoTheThird** [comment]
Confirmed on the Oneplus one. Pinging our new keyboard expert @UniversalSuperBox.

**Comment by Flohack74** [comment]
Is this regression from 15.04?

**Comment by NeoTheThird** [comment]
Well, there's no keyboard transparency in 15.04, so... yes?

**Comment by Flohack74** [comment]
Damn there isnt. So ok ;)

**Comment by mariogrip** [comment]
this is not a backlog item... this was not a thing in vivid. regression means a working thing in an older version is broken a newer one.

**Comment by Flohack74** [comment]
Ok so target it for OTA-4? I feel bad about delivering a buggy feature when we know already about it... And we can´t deny we know it xD

**Comment by mariogrip** [comment]
Humm, it seems the keyboard seem to deliver its promise of transparency, its like this because the black background does not go all the way down. I wonder if this is an toolkit item so we can edit for all apps.

**Comment by mateosalta** [comment]
Also in the browser, the content seems to resize to avoid the keyboard on the tablet.
![screenshot20180528_031519385](https://user-images.githubusercontent.com/1965279/40593027-fabf114e-61ea-11e8-8abb-cb7e7ad9f008.png)

**Comment by dpniel** [comment]
@mariogrip i agree this looks like a toolkit issue. system-settings set's `anchorToKeyboard` in its [`MainView`](https://github.com/ubports/system-settings/blob/b4bcd0f5f056fbb2286271200d6ee9ce0594a89d/src/qml/MainWindow.qml#L33) so the overlay background should be fixed to the top of the keyboard. I wonder if explicitly setting the `anchorToKeyboard` property on the pin dialog (as it also inherits `OrientationHelper`) would workaround this for now as I'm not seeing it happening with other dialogs.

@mateosalta what you are seeing is correct and was designed that way. The transparent keyboard was only really a way to make using the OSK a bit easier when in windowed mode. The norm in staged mode is to keep all content above the keyboard. But this is still left up to each app to decide if it's wants to anchor the content or not. So unfortunately we are never going to get a consistent experience here when using transparency in staged mode.
